Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,295 @@
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                try
+                {
+                    var highestAmountOfPerson = decimal.Parse(currentPersonData[1..].Max());
+                    if (highestAmountOfPerson == HighestBalanceEver)
+                    {
+                        if (i == peopleAndBalances.Length - 1)
+                        {
+                            PersonWithHighestBalance.Append(" and ");
+                        }
+                        else
+                        {
+                            PersonWithHighestBalance.Append(", ");
+                        }
+
+                        PersonWithHighestBalance.Append(currentPersonData[0]);
+                    }
+
+                    if (highestAmountOfPerson > HighestBalanceEver)
+                    {
+                        HighestBalanceEver = highestAmountOfPerson;
+                        var currentOne = PersonWithHighestBalance.ToString();
+
+                        if (string.IsNullOrEmpty(currentOne))
+                        {
+                            PersonWithHighestBalance.Append(currentPersonData[0]);
+                        }
+                        else
+                        {
+                            PersonWithHighestBalance.Replace(currentOne, currentPersonData[0]);
+                        }
+                    }
+                }
+                catch (ArgumentNullException)
+                {
+                    Console.WriteLine($"There was a person with no balance in the file, namely : {currentPersonData[0]}");
+                    break;
+                }
+            }
+
+            return HighestBalanceEver;
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (IsInValidValidInput(peopleAndBalances))
+            {
+                return InValidOutput;
+            }
+
+            var HighestLossEver = decimal.MinValue;
+            var PersonWithHighestLoss = "";
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                var allAmounts = currentPersonData[1..];
+
+                // calculateLoss
+
+                if (allAmounts.Length <= 1)
+                {
+                    return InValidOutput;
+                }
+
+                for (int j = 0; j < allAmounts.Length - 1; j++)
+                {
+                    try
+                    {
+                        var amount1 = decimal.Parse(allAmounts[j], NumberStyles.Currency | NumberStyles.Number);
+                        try
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this code to not nest more than 3 control flow statements.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4967975)

Reply:
Hmm, can someone explain that to me. I use a try with a nested if then. Or is the try also count as a if ?

Reply:
oke, I think I see the solution but not any clue how to solve this. 
Every function that I try have more then 4 parameters so please any hints

Reply:
maybe I solved it